kustomize/commands/edit/fix/convert_test.go

```diff
@@ -1143,3 +1144,76 @@ metadata:
     a.b.c: SOME_SECRET_NAME_PLACEHOLDER
 `, string(content))
 }
+
+func TestFixVarsWithPatch(t *testing.T) {
+	kustomization := []byte(`
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+patchesStrategicMerge:
+  - patch.yaml
+
+vars:
+  - name: CERTIFICATE_NAMESPACE
+    objref:
+      name: system
+    fieldref:
+      fieldpath: metadata.namespace
+`)
+	patch := []byte(`
+apiVersion: apps/v1
+kapiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
+  namespace: system
+spec:
+  template:
+    spec:
+      containers:
+        - name: $(CERTIFICATE_NAMESPACE)
+`)
+
+	fSys := filesys.MakeFsInMemory()
+	testutils_test.WriteTestKustomizationWith(fSys, kustomization)
+	fSys.WriteFile("patch.yaml", patch)
```

Comment:
lint error: `Error return value of `fSys.WriteFile` is not checked (errcheck)`

```suggestion
	assert.NoError(fSys.WriteFile("patch.yaml", patch))
```

Reply:
added the assert for all `fSys.WriteFile`